les_11_1/SnakeGame.py

```diff
@@ -78,11 +78,24 @@ def update(self, events, dt):
             self.snake = self.snake.take(self.length)
 
             self.cooldown = self.speed
-
+            
+            if self.isdead(self.snake):
+                print("you ded")
+                exit()
+            
             if self.snake.pos.is_same(self.food):
                 self.length += 1
                 snake = Snake(self.food, self.snake)
                 self.setfood()
+    
+    def isdead(self, snake):
+        headpos = snake.pos
+        snake = snake.tail
+        while snake != None:
```

Comment:
Also may be possible to make a bit more generic method as part of the Snake class to check if a snake part exists that match the given predicate.